src/Common/filesystemHelpers.cpp

```diff
@@ -130,6 +130,28 @@ bool pathStartsWith(const String & path, const String & prefix_path)
     return pathStartsWith(filesystem_path, filesystem_prefix_path);
 }
 
+String SQLiteDatabaseValidatePath(const String & path, const String & user_files_path)
```

Comment:
Common module should depend on Context or SQLite. Please consider to reuse pathStartsWith method.

Reply:
> Please consider to reuse pathStartsWith method

I did not use it on purpose, because it uses `fs::weakly_canonical` and I want `fs::canonical` as it also checks file existance.

> Common module should depend on Context or SQLite

It is a function to validate paths, so SQLite objects do not need to be passed here. Dependence on Context is not necessary.

Reply:
@kssenii it throws exception with information about SQLite. Common should not depends on any information related to sqlite.

Reply:
@kitaisreal if you need to check path existence create another support method existingPathStartsWith.